src/network/networkd-ndisc.c

```diff
@@ -1176,29 +1176,37 @@ int config_parse_address_generation_type(
         if (r < 0)
                 return log_oom();
 
-        if ((p = startswith(rvalue, "static:")))
-                token->address_generation_type = IPV6_TOKEN_ADDRESS_GENERATION_STATIC;
-        else if ((p = startswith(rvalue, "prefixstable:")))
+        if ((p = startswith(rvalue, "prefixstable"))) {
                 token->address_generation_type = IPV6_TOKEN_ADDRESS_GENERATION_PREFIXSTABLE;
-        else {
+                if (*p == ':')
+                        p++;
+                else
+                        p = NULL;
+        } else {
                 token->address_generation_type = IPV6_TOKEN_ADDRESS_GENERATION_STATIC;
-                p = rvalue;
-        }
-
-        r = in_addr_from_string(AF_INET6, p, &buffer);
-        if (r < 0) {
-                log_syntax(unit, LOG_WARNING, filename, line, r,
-                           "Failed to parse IPv6 %s, ignoring: %s", lvalue, rvalue);
-                return 0;
-        }
-
-        if (in_addr_is_null(AF_INET6, &buffer)) {
-                log_syntax(unit, LOG_WARNING, filename, line, 0,
-                           "IPv6 %s cannot be the ANY address, ignoring: %s", lvalue, rvalue);
-                return 0;
+                p = startswith(rvalue, "static:");
+                if (!p)
+                        p = rvalue;
         }
 
-        token->prefix = buffer.in6;
+        if (p) {
+                r = in_addr_from_string(AF_INET6, p, &buffer);
+                if (r < 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r,
+                                   "Failed to parse IP address in %s=, ignoring assignment: %s",
+                                   lvalue, rvalue);
+                        return 0;
+                }
+                if (token->address_generation_type == IPV6_TOKEN_ADDRESS_GENERATION_STATIC &&
+                    in_addr_is_null(AF_INET6, &buffer)) {
+                        log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                   "IPv6 address in %s= cannot be the ANY address, ignoring assignment: %s",
+                                   lvalue, rvalue);
+                        return 0;
+                }
+                token->prefix = buffer.in6;
+        } else
+                token->prefix = (struct in6_addr) {};
```

Comment:
Won't be this simper by this.
```
-- i/src/network/networkd-ndisc.c
+++ w/src/network/networkd-ndisc.c
@@ -1158,6 +1158,7 @@ int config_parse_address_generation_type(
 
         _cleanup_free_ IPv6Token *token = NULL;
         union in_addr_union buffer;
+        bool empty_prefix = false;
         Network *network = data;
         const char *p;
         int r;
@@ -1180,25 +1181,30 @@ int config_parse_address_generation_type(
                 token->address_generation_type = IPV6_TOKEN_ADDRESS_GENERATION_STATIC;
         else if ((p = startswith(rvalue, "prefixstable:")))
                 token->address_generation_type = IPV6_TOKEN_ADDRESS_GENERATION_PREFIXSTABLE;
-        else {
+        else if ((p = startswith(rvalue, "prefixstable"))) {
+                token->address_generation_type = IPV6_TOKEN_ADDRESS_GENERATION_PREFIXSTABLE;
+                empty_prefix = true;
+        } else {
                 token->address_generation_type = IPV6_TOKEN_ADDRESS_GENERATION_STATIC;
                 p = rvalue;
         }
 
-        r = in_addr_from_string(AF_INET6, p, &buffer);
-        if (r < 0) {
-                log_syntax(unit, LOG_WARNING, filename, line, r,
-                           "Failed to parse IPv6 %s, ignoring: %s", lvalue, rvalue);
-                return 0;
-        }
+        if (!empty_prefix) {
+                r = in_addr_from_string(AF_INET6, p, &buffer);
+                if (r < 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r,
+                                   "Failed to parse IPv6 %s, ignoring: %s", lvalue, rvalue);
+                        return 0;
+                }
 
-        if (in_addr_is_null(AF_INET6, &buffer)) {
-                log_syntax(unit, LOG_WARNING, filename, line, 0,
-                           "IPv6 %s cannot be the ANY address, ignoring: %s", lvalue, rvalue);
-                return 0;
-        }
+                if (in_addr_is_null(AF_INET6, &buffer)) {
+                        log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                   "IPv6 %s cannot be the ANY address, ignoring: %s", lvalue, rvalue);
+                        return 0;
+                }
 
 
-        token->prefix = buffer.in6;
+                token->prefix = buffer.in6;
+        } else
+                token->prefix = (struct in6_addr) {};
```

Reply:
It should work. But I do not see any significant difference to the current code, though...
BTW, I notice that it is not necessary to fill `token->prefix` with zero when `IPv6Token=prefixstable`, as it is already initialized in `ipv6token_new()`. I will update that.